syntax = "proto3";

package example;

option cc_generic_services = true;

message ResultCode{
  int32 error_code = 1;
  string error_message = 2;
}

message LoginRequest{
  string name = 1;
  string password = 2;
}

message LoginResponse{
  ResultCode result = 1;
  bool success = 2;
}

// 用户服务，提供用户相关的操作，如登录、注销等。
service UserServiceRpc{
  // 用户登录，验证用户名和密码，返回登录结果
  rpc Login(LoginRequest)returns(LoginResponse);
}


